sdk/python/Makefile

```diff
@@ -18,23 +18,24 @@ TEST_ALL_DEPS = build
 
 ensure::
 	$(PYTHON) -m venv venv
-	. venv/*/activate && python -m pip install -r requirements.txt
+	. venv/*/activate && pip install wheel
```

Comment:
Same problem as before, this selects the wrong pip for me:

```
[nix-shell:~/pulumi/sdk/python]$ which python
which python
/Users/anton/pulumi/sdk/python/venv/bin/python
(venv) 
[nix-shell:~/pulumi/sdk/python]$ find venv/ -name pip
find venv/ -name pip
(venv) 
echo $PATH
/Users/anton/pulumi/sdk/python/venv/bin:/nix/store/8x9677lwljl54fdlsmbqb7vws5m4f29b-bash-interactive-4.4-p23/bin:/nix/store/vn148wjd5f92yjbmlbz0rbi195daqc8a-clang-wrapper-7.1.0/bin:/nix/store/shvq72iflm8kz4qmdswhla64qypbhxjg-clang-7.1.0/bin:/nix/store/ylblikfcniagmbnhckp3fakjkbxqy4zl-coreutils-9.0/bin:/nix/store/51qq412z5d64rn85gg3nmnxwvjs1zs96-cctools-binutils-darwin-wrapper-949.0.1/bin:/nix/store/854jyvxrvpdpbfn2zaba1v2qgqkxipyh-cctools-binutils-darwin-949.0.1/bin:/nix/store/8f99z1hxrqvapgxjhgxi4qmf2ms38rzn-go-1.17.4/bin:/nix/store/8mzh4l16kll8ldcawgkwl9bwxpbayawb-xcodebuild-0.1.2-pre/bin:/nix/store/i7liimm017jj158lh2ac51b1s6ixg639-Toolchains/XcodeDefault.xctoolchain/bin:/nix/store/rqbzwkyvjz5lvzqhxm6khi4pr44451rn-curl-7.79.1-dev/bin:/nix/store/sia74dxywyll3wxf36r8cx45fsr0qrjx-nghttp2-1.43.0-bin/bin:/nix/store/spbzg88dcr2b0rkb1j4gzxs5y09ggsy9-libidn-1.38-bin/bin:/nix/store/0rpamhzf29jswrjydgr6qf77v7r95d7y-libkrb5-1.18-dev/bin:/nix/store/f4qs0hfxgimwgi3qp96il5yhy0faxd90-libkrb5-1.18/bin:/nix/store/l6d63k59m5cvwd2iq8ma1f5509czwda3-openssl-1.1.1l-bin/bin:/nix/store/k8crx6gs0xgxcnnaksc0ldg3bxzvnlcb-brotli-1.0.9/bin:/nix/store/5p65pd7cnr8a1qkhm802bj28qpnnajaq-curl-7.79.1-bin/bin:/nix/store/6fxw0ndpxrpp5f5vz08f955jr4lvib08-delve-1.7.3/bin:/nix/store/pvi01zkzzf5wicivi8wdryy0wh97vynf-dotnet-sdk-3.1.415/bin:/nix/store/n8c296ds88azm5gs1vdrmcsacpqi3k8s-gcc-wrapper-10.3.0/bin:/nix/store/vicydk3qfcjwq0m9lpwn3i9cw0g7g70j-gcc-10.3.0/bin:/nix/store/bcjf7pr923hinmzcci9qh47g4zd3ds4q-git-2.34.0/bin:/nix/store/y0alig52a3n6n8l6rwxwcb61y2yci7p9-golangci-lint-1.43.0/bin:/nix/store/bai232dkgy6ynm7ikf7h64gbwrnncfk9-goreleaser-1.1.0/bin:/nix/store/iwggnay7nwxq0jlq3xhzdia5yy4mpyjx-typescript-4.5.2/bin:/nix/store/w5zibyy9fnziqk0b9gvqzpq0zcg0s3xi-nodejs-14.18.1/bin:/nix/store/v7wb8dq7y0bbvyhw3va9j80985ark48l-pipenv-2021.11.23/bin:/nix/store/w8lqcsrwlwrvmchj46ph9880w9h4m50a-python3-3.9.6/bin:/nix/store/h5j6wymds31fgj0pdhfrdh27qhlw6d1v-python3.9-pip-21.1.3/bin:/nix/store/s1n1zn5x3mikzbmvg3gxwqk47ndni81f-python3.9-virtualenv-20.8.1/bin:/nix/store/7sfnv99k41mdr21a558k8c4g1gd2zqz3-python3.9-virtualenv-clone-0.5.7/bin:/nix/store/ws85f4b0nzwiz9lhn2ynsi0slyajqsk7-which-2.21/bin:/nix/store/55rzqqfq6kq59py3il6azfjy2zm2jys9-yarn-1.22.17/bin:/nix/store/c0vib3nvcwws508bb2y79ndifdbds03d-pulumictl-0.0.28/bin:/nix/store/ylblikfcniagmbnhckp3fakjkbxqy4zl-coreutils-9.0/bin:/nix/store/6bw255bhrkavpm80gwbi3qc4y0c3y8yv-findutils-4.8.0/bin:/nix/store/1iv4czqa4yic4sbl2lf7z1yygrlwpp5a-diffutils-3.8/bin:/nix/store/kb2f11i8ahynwp0hz7yhg201v0n21b7x-gnused-4.8/bin:/nix/store/gl3k1v2ghjh7sg2v0gdb3qmm2yfyi35p-gnugrep-3.7/bin:/nix/store/nvk26mizxlrlpxacchab74ha77ag0x7r-gawk-5.1.1/bin:/nix/store/b39m837bsf6b50aqd0n3s6maar8aqk6a-gnutar-1.34/bin:/nix/store/pjxzcxpgf6whynjgxadzbpd65icyblm3-gzip-1.11/bin:/nix/store/nw6wszn70is36w8g1b6nxhn8injz6yky-bzip2-1.0.6.0.2-bin/bin:/nix/store/2d9zk8y6g0k8zzjyqgbdv2wiaq228yyk-gnumake-4.3/bin:/nix/store/3npg6a8nc5vpcyw98v085cmlz7f78kgs-bash-5.1-p12/bin:/nix/store/5pwyalfadcnrdnipazywfkc9dnza9ag1-patch-2.7.6/bin:/nix/store/nyc498wmc43021s3gdy8f6vvddbhg5iw-xz-5.2.5-bin/bin:/Users/anton/bin:/Users/anton/.nix-profile/bin:/usr/bin:/bin:/usr/sbin:/sbin
```

I think this is because venv does not contain pip executable.

```
ls venv/bin/
Activate.ps1  activate	activate.csh  activate.fish  python  python3  python3.9
```

```
[nix-shell:~/pulumi/sdk/python]$ python --version
python --version
Python 3.9.6
(venv) 
```

I suggest we revert this particular change unless there's reasons why `python -m pip` is problematic in some way?

I did not have pip installed globally so I tried installing it globally but then the `pip` command attempts a global install of the reqs and fails, it does not install the reqs into the VM.


```
make ensure
ENSURE:
python3 -m venv venv
. venv/*/activate && pip install wheel
Collecting wheel
  Using cached wheel-0.37.1-py2.py3-none-any.whl (35 kB)
Installing collected packages: wheel
ERROR: Could not install packages due to an OSError: [Errno 13] Permission denied: '/nix/store/w8lqcsrwlwrvmchj46ph9880w9h4m50a-python3-3.9.6/lib/python3.9/site-packages/wheel'
Consider using the `--user` option or check the permissions.

WARNING: You are using pip version 21.1.3; however, version 21.3.1 is available.
You should consider upgrading via the '/nix/store/w8lqcsrwlwrvmchj46ph9880w9h4m50a-python3-3.9.6/bin/python3.9 -m pip install --upgrade pip' command.
make: *** [Makefile:21: ensure] Error 1
(venv) 
```

Reply:
I'm wondering in your environment which is probably more common than mine (1) `which pip` outside of venv (2) `which pip` inside venv (3) can we make sure the packages pip installs are installed in the local not global environment?

Somehow this works on the CI environment judging by passing tests.

Reply:
@t0yv0 interesting. I agree w/ this revert and wonder what's different between our environments and Python's `-m venv` behavior.

Reply:
@t0yv0 reverted to using `python -m pip` and validated that this ran on Windows & Linux